src/grisp_cryptoauth_api_server.erl

```diff
@@ -27,11 +27,24 @@ handle_call({Fun, Args}, _From, {Context, OldTRef}) ->
     %% to sleep after SLEEP_TIME_SEC seconds to save energy
     timer:cancel(OldTRef), %% this doesn't throw on bad args
     {ok, NewTRef} = timer:apply_after(?SLEEP_TIME_SEC * 1000, grisp_cryptoauth, sleep, [Context]),
-    try apply(grisp_cryptoauth, Fun, [Context | Args]) of
-        Result -> {reply, {ok, Result}, {Context, NewTRef}}
-    catch
-        C:R:S -> {reply, {error, C, R, S}, {Context, NewTRef}}
-    end.
+    {reply, retry(5, 300, Fun, [Context | Args]), {Context, NewTRef}}.
 
 handle_cast(_, State) ->
     {noreply, State}.
+
+retry(RetryCount, SleepTime, Fun, Args) ->
```

Comment:
Hmm not even DEBUG logs?

Reply:
The API server is not logging anything, so I wan't sure about adding log...
If you think so I could add debug logging...

Reply:
I think it would be nice so we could still observe the bug